copy edit pt. 2: Statements topics (#92)
microsoft/qsharp-language: Specifications/Language/2_Statements/ConditionalLoops.md

# Conditional loops

> Like most classical programming languages, Q# supports loops that break based on a condition: loops for which the number of iterations is unknown and may vary from run to run. Since the instruction sequence is unknown at compile-time, you need to handle these conditional loops with particular care in a quantum runtime. 

Author: ```suggestion
Like most classical programming languages, Q# supports loops that break based on a condition: loops for which the number of iterations is unknown and may vary from run to run. Since the instruction sequence is unknown at compile-time, the compiler handles these conditional loops in a particular way in a quantum runtime. 
```

> Like most classical programming languages, Q# supports loops that break based on a condition: loops for which the number of iterations is unknown and may vary from run to run. Since the instruction sequence is unknown at compile-time, you need to handle these conditional loops with particular care in a quantum runtime. 

Author: I think what threw me off here was the phrase "handled with particular care" - that's a subjective quality for a compiler to have.

> As long as the condition does not depend on quantum measurements, conditional loops can be handled without issues by doing a just-in-time compilation before sending the instruction sequence to the quantum processor. 

Author: ```suggestion
As long as the condition does not depend on quantum measurements, conditional loops are processed with a just-in-time compilation before sending the instruction sequence to the quantum processor. 
```
